components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+	const numDays = getNumberOfDaysInMonth(shownMonth, shownYear);
+
+	// populate first week of month
+	const firstWeek = [];
+	const numDaysFromPrevMonthToShow = getNumberOfDaysFromPrevMonthToShow(shownMonth, shownYear);
+
+	if (numDaysFromPrevMonthToShow > 0) {
+		const prevMonth = getPrevMonth(shownMonth);
+		const prevMonthYear = (shownMonth === 0) ? (shownYear - 1) : shownYear;
+		const numDaysLastMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+		for (let i = numDaysLastMonth - numDaysFromPrevMonthToShow + 1; i <= numDaysLastMonth; i++) {
+			firstWeek.push(new Date(prevMonthYear, prevMonth, i));
+		}
+	}
+	for (let j = 1; j <= daysInWeek - numDaysFromPrevMonthToShow; j++) {
+		firstWeek.push(new Date(shownYear, shownMonth, j));
+	}
+	dates.push(firstWeek);
+
+	// remaining weeks
+	let nextMonthDay = 1;
+	let firstDateOfWeek = daysInWeek - numDaysFromPrevMonthToShow + 1;
+	const numWeeks = Math.ceil((numDaysFromPrevMonthToShow + numDays) / daysInWeek);
+	const nextMonth =  getNextMonth(shownMonth);
+	const nextMonthYear = (shownMonth === 11) ? (shownYear + 1) : shownYear;
+	for (let i = 1; i < numWeeks; i++) {
+		const week = [];
+		for (let j = firstDateOfWeek; j < firstDateOfWeek + daysInWeek; j++) {
+			let day;
+			if (j > numDays) {
+				day = new Date(nextMonthYear, nextMonth, nextMonthDay);
+				nextMonthDay++;
+			} else {
+				day = new Date(shownYear, shownMonth, j);
+			}
+			week.push(day);
+		}
+		firstDateOfWeek = firstDateOfWeek + daysInWeek;
+		dates.push(week);
+	}
+
+	return dates;
+}
+
+export function getNextMonth(month) {
+	return (month === 11) ? 0 : (month + 1);
+}
+
+export function getNumberOfDaysFromPrevMonthToShow(month, year) {
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	let numDaysFromLastMonthToShowThisMonth = 0;
+	if (firstDayOfMonth !== firstDayOfWeek) {
+		numDaysFromLastMonthToShowThisMonth = firstDayOfMonth - firstDayOfWeek;
+		if (numDaysFromLastMonthToShowThisMonth < 0) {
+			numDaysFromLastMonthToShowThisMonth += daysInWeek;
+		}
+	}
+	return numDaysFromLastMonthToShowThisMonth;
+}
+
+export function getNumberOfDaysInMonth(month, year) {
+	return new Date(year, month + 1, 0).getDate();
+}
+
+export function getNumberOfDaysToSameWeekPrevMonth(month, year) {
+	const prevMonth = getPrevMonth(month);
+	const prevMonthYear = prevMonth === 11 ? year - 1 : year;
+	const numDaysPrevMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+	const numDaysFromPrevPrevMonthToShowPrevMonth = getNumberOfDaysFromPrevMonthToShow(prevMonth, prevMonthYear);
+	const numWeeksPrevMonth = Math.ceil((numDaysFromPrevPrevMonthToShowPrevMonth + numDaysPrevMonth) / 7);
+
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	const hasDaysFromPrevMonth = (firstDayOfMonth !== firstDayOfWeek);
+	return daysInWeek * (numWeeksPrevMonth - ((hasDaysFromPrevMonth) ? 1 : 0));
+}
+
+export function getPrevMonth(month) {
+	return (month === 0) ? 11 : (month - 1);
+}
+
+export function parseDate(val) {
```

Comment:
I feel like this isn't great and I should go about this differently and should incorporate `convertUTCToLocalDateTime` to determine what date the user should actually see

Reply:
Hmm, yeah worth having a discussion about as this will impact the calendar, time and date inputs. These ISO formats are always UTC offset zero, so yeah... maybe we should indeed be converting to local (if available) before rendering and then back to UTC when the value changes. I like that...

Reply:
If this is the case, then we always need the time as well, ya?

Reply:
So I _think_ this should now work with other timezones. It will now parse the ISO date time string, convert to local, and then use local as `selectedValue`. I think the rest of the code should be fine. For example, If ISO string is `2019-08-01T20:00:00Z` and timezone is in Australia, then we would expect selected value to be August 2 (which seems to work), so we have `parseDate` return that, and the focus date becomes that, etc. When a value is selected, it would then `selectedValue` to convert from local date to UTC.

The weird thing I'm not sure about here is the affect of time on output/changing `selectedValue`. I have `calendar.js` currently using 12 as the hour when creating a date to keep things consistent (though maybe 0 is a better choice) but this would potentially be affected by time picker as well. Not sure if this should be considered as part of this PR or in a subsequent one when the date/time work starts to combine.

Hopefully this all made sense. Timezones seem to get confusing (at least to me!)